chore: migrate to vitest

caio2983 opened the pull request:
#947 

Hello @artalar ,
Please let me know if any changes need to be done in this

caio2983 committed:
chore: migrate async package from uvu to vitest

caio2983 committed:
chore: migrate core package from uvu to vitest

caio2983 committed:
chore: migrate core-v1 package from uvu to vitest

caio2983 committed:
chore: migrate core-v2 package from uvu to vitest

caio2983 committed:
chore: migrate effects package from uvu to vitest

caio2983 committed:
chore: migrate framework package from uvu to vitest

caio2983 committed:
chore: migrate hooks package from uvu to vitest

caio2983 committed:
chore: migrate jsx package from uvu to vitest

caio2983 committed:
chore: migrate lens package from uvu to vitest

caio2983 committed:
chore: migrate logger package from uvu to vitest

caio2983 committed:
chore: migrate npm-cookie-baker package from uvu to vitest

caio2983 committed:
chore: migrate npm-lit package from uvu to vitest

caio2983 committed:
chore: migrate npm-react package from uvu to vitest

caio2983 committed:
chore: migrate npm-solid-js package from uvu to vitest

caio2983 committed:
chore: migrate npm-svelte package from uvu to vitest

caio2983 committed:
chore: migrate npm-vue package from uvu to vitest

caio2983 committed:
chore: migrate npm-zod package from uvu to vitest

caio2983 committed:
chore: migrate persist package from uvu to vitest

caio2983 committed:
chore: migrate persist-web-storage package from uvu to vitest

caio2983 committed:
chore: migrate primitive package from uvu to vitest

caio2983 committed:
chore: migrate testing package from uvu to vitest

caio2983 committed:
chore: migrate timer package from uvu to vitest

caio2983 committed:
chore: migrate undo package from uvu to vitest

caio2983 committed:
chore: migrate url package from uvu to vitest

caio2983 committed:
chore: migrate utils package from uvu to vitest

caio2983 committed:
chore: migrate web package from uvu to vitest

caio2983 committed:
chore: migrate devtools package from uvu to vitest

caio2983 committed:
chore: migrate form package from uvu to vitest

caio2983 committed:
chore: migrate form-web package from uvu to vitest

artalar commented on packages/async/src/index.story.test.ts:
```diff
@@ -13,9 +12,9 @@ describe('optimistic update', () => {
-    Also, we use `onConnect` to fetch the list from the server every 5 seconds
-    and we don't want to call subscriptions extra times so we use `isDeepEqual`
-    in `withDataAtom` to prevent new reference stream if nothing really changed.
+    Also, we use onConnect to fetch the list from the server every 5 seconds
```
Why the quotes removed?

artalar commented on packages/async/src/index.story.test.ts:
```diff
@@ -30,25 +29,16 @@ describe('optimistic update', () => {
   const INTERVAL = 5
 
-  const getData = reatomAsync.from(api.getData).pipe(
-    // add `dataAtom` and map the effect payload into it
```
Why the comments removed?

artalar commented on packages/async/src/mapToAsync.test.ts:
```diff
@@ -1,55 +1,52 @@
 import { atom } from '@reatom/core'
 import { mapToAsync, withDataAtom } from './index'
 
-export const test = suite('mapToAsync')
```
There is no need to use `describe`, please change the tests runner script to this:

```json
    "test": "vitest run src/*.test.ts",
    "test:watch": "vitest src/*.test.ts"
```

artalar reviewed (changes requested):
Huge work, thank you! Please, follow the comments and be ready that today I will publish a couple updates and refactoring INCLUDING the async package migration to the vitest, to there will be a small merge conflicts

caio2983 committed:
fix(async): retrieve removed comments

caio2983 committed:
refactor(async): remove describe usage

caio2983 committed:
refactor(async): change tests runner script

caio2983 commented:
> Huge work, thank you! Please, follow the comments and be ready that today I will publish a couple updates and refactoring INCLUDING the async package migration to the vitest, to there will be a small merge conflicts

Understood, I just removed that one describe usage , changed the tests runner script and fixed the comments . 

Should I change the tests runner scripts to     "test": "vitest run src/*.test.ts",
    "test:watch": "vitest src/*.test.ts" in every package besides async?

artalar commented:
@caio2983 ive updated the deps and the async package tests, please check

> Should I change the tests runner scripts to "test": "vitest run src/.test.ts",
"test:watch": "vitest src/.test.ts" in every package besides async?

yep

caio2983 committed:
chore: change test scripts and remove non-english comments

caio2983 committed:
chore: migrate web-fetch package from uvu to vitest

artalar commented:
@caio2983 Hi! Sorry for the delay, I have been super busy, working, sick and many other bus factor things :) Are you interested in continuing to work on this PR? I would love to merge it as soon as possible.

AgarkovRoman commented on packages/async/src/index.story.test.ts:
```diff
@@ -1,5 +1,4 @@
-import { test } from 'uvu'
-import * as assert from 'uvu/assert'
+import { describe, it, expect } from 'vitest'
```
btw, you can set up a `global: true` in vitest.config and this will allow you to remove all imports from 'vitest'

https://vitest.dev/config/#globals

artalar replied:
imho direct import is more obvious.

AgarkovRoman commented:
consider to use a [workspace feature](https://vitest.dev/guide/workspace.html#workspace) in vitest which can be useful for monorepo such as reatom

AgarkovRoman commented:
@artalar I can take over this and continue migration, do you mind?

artalar commented:
@AgarkovRoman ok, go ahead.

artalar commented:
done in #1026
